tycho-pomgenerator-plugin/src/main/java/org/eclipse/tycho/pomgenerator/AbstractGeneratePomsMojo.java

```diff
@@ -0,0 +1,78 @@
+package org.eclipse.tycho.pomgenerator;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.maven.model.Model;
+import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
+import org.apache.maven.model.io.xpp3.MavenXpp3Writer;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.codehaus.plexus.util.ReaderFactory;
+import org.codehaus.plexus.util.xml.XmlStreamReader;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+
+public abstract class AbstractGeneratePomsMojo extends AbstractMojo {
+
+    /**
+     * Location of directory with template pom.xml file. pom.xml templates will be looked at this
+     * directory first, default templates will be used if template directory and the template itself
+     * does not exist.
+     * 
+     * See src/main/resources/templates for the list of supported template files.
+     */
+    @Parameter(property = "templatesDir", defaultValue = "${basedir}/pom-templates")
+    private File templatesDir;
```

Comment:
If I understand the code in org.apache.maven.archiver.MavenArchiver.createArchive(MavenSession, MavenProject, MavenArchiveConfiguration) corectly it is supposed that the pom itself is included.

What do you think instead of reading a template, simply reading the pom.xml itself and extending those?

Reply:
So in case of a pomless build, this would simply point to .polyglot.META-INF, right?
I decided against it, because it contains the <parent>...</parent> entry, pointing to the connector pom, which I don't want to keep in the metadata.
Otherwise the existing pom is almost identical to the one I create. And since it is already there, reusing it sounds like a good idea.

Reply:
If you want to support pomless the correct way to obtain the pom-**model** is to use something like:

```
List<ModelProcessor> lookupList = container.lookupList(ModelProcessor.class);
            for (ModelProcessor processor : lookupList) {
                File pom = processor.locatePom(projectBaseFolder);
                if (pom != null && pom.exists()) {
                    Map<String, Object> options = new HashMap<>(1);
                    options.put(ModelProcessor.SOURCE, new FileModelSource(pom));
                    Model project model = parentPom.getReader().read(pom, options);
                }
            }
```

of course you then could strip unwanted content from the model. To write it out you can use `ModelWriter`

`writer = container.lookup(ModelWriter.class, "xml");`

Reply:
Beside that it might be better to use `org.apache.maven.project.MavenProject.getModel()` as a basis e.g. write it out, then read it again with `ModelReader`, modify it and write the result again...

Reply:
Its even easier: The project.getFile() seems to always point to the "real" pom file. In case of pomless to the `.polyglot.META-INF` file so it seems one could simply use this file (if present) as a base for the generated pom.